### PR TITLE
Move c module to /msprime

### PR DESCRIPTION
**Pull request by benjeffery**
Needed for #1129, separated out for discussion.

The main breakage will be if users are doing `import _msprime` but as that is considered internal that might be acceptable.

### CODE REVIEW

File: tests/test_dict_encoding.py [hunk 1]
```diff
@@ -11,8 +11,8 @@
 import numpy as np
 import tskit
 
-import _msprime as c_module
```
Comment (jeromekelleher) [inline tests/test_dict_encoding.py:14]: Let's change this to ``import msprime._msprime as c_module``. This'll make the diffs to the tskit version smaller and make our lives easier.

**Review by jeromekelleher — CHANGES_REQUESTED** [review]
I thought we might have some stuff in the documentation that would be misleading, but it looks like we're all OK. This is quite an improvement, actually, keeping all the msprime stuff withing the package directory. Excellent!

One minor change above.

File: tests/test_dict_encoding.py (hunk 1 above)
Comment (benjeffery) [reply tests/test_dict_encoding.py:14]: Done!

**Comment by benjeffery** [comment]
@jeromekelleher changes made, should be good to go.

**Review by jeromekelleher — APPROVED** [review]

**Pull request merged**